pkg/workqueue/queue.go

```diff
@@ -69,19 +78,36 @@ func NewWithConfig(name string, config Config) Interface {
 				), config.OverallRateLimiterMaxDelay),
 			workqueue.TypedRateLimitingQueueConfig[string]{
 				Name: name,
+				DelayingQueue: workqueue.NewTypedDelayingQueueWithConfig(workqueue.TypedDelayingQueueConfig[string]{
+					Name: name,
+					Queue: workqueue.NewTypedWithConfig(workqueue.TypedQueueConfig[string]{
+						Name:  name,
+						Queue: &priorityWorkQueue[string]{priorityQueue: priorityQueue},
+					}),
+				}),
 			}),
 		name: name,
 	}
 }
 
-func (q *queueType) Enqueue(obj interface{}) {
+func (q *queueType) enqueue(obj interface{}, priority int) {
```

Comment:
Could the priority use a specific type? It would make it more obvious that this expects one of the two constants, and not an arbitrary priority. (This isn’t all that important since the method is private.)

Reply:
We could but like you said it's private. I only defined exposed and priorities (via specific functions) at least for now. Perhaps we'll have a future need for users to specify arbitrary priorities in which case a specific type wouldn't make sense. I figure we can deal with that if/when. For now I kept it simple with "low" and "normal".

Reply:
Another approach is to move the low and normal priority concept internally to the resource syncer and have the work queue take an arbitrary priority, ie change `EnqueueLowPriority` to `EnqueueWithPriority(...int priority)`.

Reply:
I changed the method to `EnqueueWithPriority` and also added a `rateLimiting` flag. Looking at the K8s controller-runtime code, they do not enqueue items from the initial listing on startup with rate limiting which makes sense b/c it already knows it's a one-time burst.  I did the same in the resource syncer.